admin_organization.go

```diff
@@ -0,0 +1,175 @@
+type AdminOrganizations interface {
+
+	// List the module sharing partnerships that an organization has
+	ListModuleConsumers(ctx context.Context, organization string) (*OrganizationList, error)
+
+	// Update the module sharing consumers that an organization has
+	UpdateModuleConsumers(ctx context.Context, organization string, consumers ModuleConsumers) (*OrganizationList, error)
+
+	// Read attributes of an existing organization via admin API.
+	Read(ctx context.Context, organization string) (*AdminOrganization, error)
+
+	// Update attributes of an existing organization via admin API.
+	Update(ctx context.Context, organization string, options AdminOrganizationUpdateOptions) (*AdminOrganization, error)
+
+	// Delete an organization by its name via admin API
+	Delete(ctx context.Context, organization string) error
+}
+
+// adminOrganizations implements AdminOrganizations.
+type adminOrganizations struct {
+	client *Client
+}
+
+// ModuleConsumers is a typed list of organization names to represent the
+// list of module consumers for updates
+type ModuleConsumers []*string
+
+// modulePartnershipUpdateOption represents the option for updating an organization's module sharing partnerships
+// this is private as it's used internally
+type modulePartnershipUpdateOption struct {
+	ID *string `jsonapi:"primary,organizations"`
+}
+
+// AdminOrganization represents a Terraform Enterprise organization returned from the Admin API.
+type AdminOrganization struct {
+	Name string `jsonapi:"primary,organizations"`
+
+	AccessBetaTools                  bool   `jsonapi:"attr,access-beta-tools"`
+	ExternalID                       string `jsonapi:"attr,external-id"`
+	GlobalModuleSharing              bool   `jsonapi:"attr,global-module-sharing"`
+	IsDisabled                       bool   `jsonapi:"attr,is-disabled"`
+	NotificationEmail                string `jsonapi:"attr,notification-email"`
+	SsoEnabled                       bool   `jsonapi:"attr,sso-enabled"`
+	TerraformBuildWorkerApplyTimeout string `jsonapi:"attr,terraform-build-worker-apply-timeout"`
+	TerraformBuildWorkerPlanTimeout  string `jsonapi:"attr,terraform-build-worker-plan-timeout"`
+}
+
+// AdminOrganizationUpdateOptions represents the admin options for updating an organization.
+// https://www.terraform.io/docs/cloud/api/admin/organizations.html#request-body
+type AdminOrganizationUpdateOptions struct {
+	AccessBetaTools                  *bool   `jsonapi:"attr,access-beta-tools,omitempty"`
+	GlobalModuleSharing              *bool   `jsonapi:"attr,global-module-sharing,omitempty"`
+	IsDisabled                       *bool   `jsonapi:"attr,is-disabled,omitempty"`
+	TerraformBuildWorkerApplyTimeout *string `jsonapi:"attr,terraform-build-worker-apply-timeout,omitempty"`
+	TerraformBuildWorkerPlanTimeout  *string `jsonapi:"attr,terraform-build-worker-plan-timeout,omitempty"`
+}
+
+func (s *adminOrganizations) ListModuleConsumers(ctx context.Context, organization string) (*OrganizationList, error) {
```

Comment:
Does the ListModuleConsumers support sending pagination info in the request? I think the default is to just return 20 records.

Reply:
It does, adding a test to support the pagination APIs